haproxy-ingress/templates/controller-servicemonitor.yaml

```
@@ -23,10 +23,26 @@ spec:
     params:
       {{- toYaml .Values.controller.serviceMonitor.params | nindent 6 }}
     {{- end }}
+    {{- if .Values.controller.serviceMonitor.metricRelabelings }}
+    metricRelabelings:
+      {{- toYaml .Values.controller.serviceMonitor.metricRelabelings | nindent 6 }}
+    {{- end }}
+    {{- if .Values.controller.serviceMonitor.relabelings }}
+    relabelings:
+      {{- toYaml .Values.controller.serviceMonitor.relabelings | nindent 6 }}
+    {{- end }}
   - honorLabels: {{ .Values.controller.serviceMonitor.honorLabels }}
     interval: {{ .Values.controller.serviceMonitor.interval }}
     path: /metrics
     port: ctrl-metrics
+    {{- if .Values.controller.serviceMonitor.metricRelabelings }}
+    metricRelabelings:
+      {{- toYaml .Values.controller.serviceMonitor.metricRelabelings | nindent 6 }}
+    {{- end }}
```

Controller metric relabeling needs to be separated from the haproxy metrics. Since they have completely different sets of metrics. We don't want to apply one to both.

```suggestion
    {{- if .Values.controller.serviceMonitor.ctrlMetricRelabelings }}
    metricRelabelings:
      {{- toYaml .Values.controller.serviceMonitor.ctrlMetricRelabelings | nindent 6 }}
    {{- end }}
```